### ISSUE TITLE
Simplify embed/weblib code with Promise and async/await

### ISSUE THREAD

**Issue by mihailik** [jeffpar/pcjs.v1#65]
Those make code way simpler to track and follow, and can be transcoded back to ES5 safely.

**Comment by jeffpar** [comment]
True, any async code I wrote pre-dates async/await (and probably isn't very well written by today's standards), but as long as it's not broken, it's low on my mental list of priorities.  Not to say I wouldn't happily accept rewrites of any old crappy code.
